In [42]:
import pandas as pd
df = pd.read_excel("./app_build/MDSInc_sales.xlsx", sheet_name="sales")
df.head()

order_id           order_date            ship_date       ship_mode  \
0     AG-2011-2040  2011-01-01 00:00:00  2011-06-01 00:00:00  Standard Class   
1    IN-2011-47883  2011-01-01 00:00:00  2011-08-01 00:00:00  Standard Class   
2     HU-2011-1220  2011-01-01 00:00:00  2011-05-01 00:00:00    Second Class   
3  IT-2011-3647632  2011-01-01 00:00:00  2011-05-01 00:00:00    Second Class   
4    IN-2011-47883  2011-01-01 00:00:00  2011-08-01 00:00:00  Standard Class   

     customer_name      segment            state    country  market   region  \
0  Toby Braunhardt     Consumer      Constantine    Algeria  Africa   Africa   
1      Joseph Holt     Consumer  New South Wales  Australia    APAC  Oceania   
2    Annie Thurman     Consumer         Budapest    Hungary    EMEA     EMEA   
3     Eugene Moren  Home Office        Stockholm     Sweden      EU    North   
4      Joseph Holt     Consumer  New South Wales  Australia    APAC  Oceania   

         product_id         category sub_category  \
0  OFF-TEN-10000025  Office Supplies      Storage   
1   OFF-SU-10000618  Office Supplies     Supplies   
2  OFF-TEN-10001585  Office Supplies      Storage   
3   OFF-PA-10001492  Office Supplies        Paper   
4   FUR-FU-10003447        Furniture  Furnishings   

                  product_name  sales  quantity  discount   profit  \
0          Tenex Lockers, Blue    408         2       0.0  106.140   
1     Acme Trimmer, High Speed    120         3       0.1   36.036   
2      Tenex Box, Single Width     66         4       0.0   29.640   
3  Enermax Note Cards, Premium     45         3       0.5  -26.055   
4   Eldon Light Bulb, Duo Pack    114         5       0.1   37.770   

   shipping_cost order_priority  
0          35.46         Medium  
1           9.72         Medium  
2           8.17           High  
3           4.82           High  
4           4.70         Medium

In [43]:
df['category'].unique()

array(['Office Supplies', 'Furniture', 'Technology'], dtype=object)

In [44]:
furniture = df.loc[df['category'] == 'Furniture']
furniture

order_id           order_date            ship_date  \
4        IN-2011-47883  2011-01-01 00:00:00  2011-08-01 00:00:00   
9         IZ-2011-4680  2011-03-01 00:00:00  2011-07-01 00:00:00   
11       IN-2011-65159  2011-03-01 00:00:00  2011-07-01 00:00:00   
12     ES-2011-4869686  2011-03-01 00:00:00  2011-07-01 00:00:00   
14       ID-2011-80230  2011-03-01 00:00:00  2011-09-01 00:00:00   
...                ...                  ...                  ...   
33725    IN-2013-52972           31-12-2013  2014-04-01 00:00:00   
33728    IN-2013-83947           31-12-2013  2014-05-01 00:00:00   
33735   MX-2013-105046           31-12-2013  2014-01-01 00:00:00   
33746     TU-2013-9400           31-12-2013  2014-04-01 00:00:00   
33754     TU-2013-9400           31-12-2013  2014-04-01 00:00:00   

            ship_mode      customer_name    segment             state  \
4      Standard Class        Joseph Holt   Consumer   New South Wales   
9      Standard Class   Lindsay Williams  Corporate            Ninawa   
11       Second Class       Larry Blacks   Consumer  National Capital   
12     Standard Class  Dorothy Dickinson   Consumer           England   
14     Standard Class       Ken Lonsdale   Consumer          Auckland   
...               ...                ...        ...               ...   
33725  Standard Class    Brian DeCherney   Consumer       Maharashtra   
33728  Standard Class      Maureen Gnade   Consumer        Queensland   
33735     First Class           Amy Hunt   Consumer              León   
33746  Standard Class      Tony Molinari   Consumer         Gaziantep   
33754  Standard Class      Tony Molinari   Consumer         Gaziantep   

              country market          region        product_id   category  \
4           Australia   APAC         Oceania   FUR-FU-10003447  Furniture   
9                Iraq   EMEA            EMEA  FUR-NOV-10002791  Furniture   
11        Philippines   APAC  Southeast Asia   FUR-TA-10002797  Furniture   
12     United Kingdom     EU           North   FUR-BO-10000728  Furniture   
14        New Zealand   APAC         Oceania   FUR-CH-10000214  Furniture   
...               ...    ...             ...               ...        ...   
33725           India   APAC    Central Asia   FUR-BO-10001641  Furniture   
33728       Australia   APAC         Oceania   FUR-BO-10001746  Furniture   
33735       Nicaragua  LATAM         Central   FUR-FU-10003608  Furniture   
33746          Turkey   EMEA            EMEA  FUR-NOV-10003257  Furniture   
33754          Turkey   EMEA            EMEA  FUR-ELD-10001413  Furniture   

      sub_category                               product_name  sales  \
4      Furnishings                 Eldon Light Bulb, Duo Pack    114   
9           Chairs           Novimex Swivel Stool, Set of Two    667   
11          Tables  Chromcraft Round Table, Adjustable Height    211   
12       Bookcases         Dania Corner Shelving, Traditional    854   
14          Chairs              Hon Rocking Chair, Set of Two    159   
...            ...                                        ...    ...   
33725    Bookcases          Bush 3-Shelf Cabinet, Traditional    143   
33728    Bookcases                Dania 3-Shelf Cabinet, Pine    171   
33735  Furnishings                 Advantus Light Bulb, Black     40   
33746       Chairs               Novimex Chairmat, Adjustable     24   
33754  Furnishings                    Eldon Light Bulb, Black     10   

       quantity  discount    profit  shipping_cost order_priority  
4             5      0.10   37.7700           4.70         Medium  
9             4      0.00  253.3200          81.26           High  
11            1      0.55  -70.3995          21.32           High  
12            7      0.00  290.4300          12.56         Medium  
14            2      0.40  -95.6760          10.07            Low  
...         ...       ...       ...            ...            ...  
33725         1      0.00   50.1600          10.46    

In [45]:
cols = ['order_id','ship_date','ship_mode','customer_name','segment','state','country','market','region','product_id','category','sub_category','product_name','quantity','discount','profit','shipping_cost','order_priority']
furniture.drop(cols,axis=1,inplace=True)
furniture

/Users/maichienvithien/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



order_date  sales
4      2011-01-01 00:00:00    114
9      2011-03-01 00:00:00    667
11     2011-03-01 00:00:00    211
12     2011-03-01 00:00:00    854
14     2011-03-01 00:00:00    159
...                    ...    ...
33725           31-12-2013    143
33728           31-12-2013    171
33735           31-12-2013     40
33746           31-12-2013     24
33754           31-12-2013     10

[6529 rows x 2 columns]

In [46]:
furniture['order_date'] = pd.to_datetime(furniture['order_date'])
furniture = furniture.sort_values('order_date')
print(furniture)

      order_date  sales
4     2011-01-01    114
439   2011-01-02    310
434   2011-01-02    291
830   2011-01-03    352
812   2011-01-03   1314
...          ...    ...
33715 2013-12-31    499
33712 2013-12-31    171
33709 2013-12-31    706
33723 2013-12-31    180
33754 2013-12-31     10

[6529 rows x 2 columns]


<ipython-input-46-80f8fc29bc8d>:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [47]:
#Now let's find how much total furniture sales occurred on each date
furniture = furniture.groupby('order_date')['sales'].sum().reset_index()
furniture = furniture.set_index('order_date')

In [48]:
furniture.index

DatetimeIndex(['2011-01-01', '2011-01-02', '2011-01-03', '2011-01-04',
               '2011-01-06', '2011-01-07', '2011-01-08', '2011-01-09',
               '2011-01-11', '2011-01-12',
               ...
               '2013-12-20', '2013-12-21', '2013-12-23', '2013-12-24',
               '2013-12-25', '2013-12-26', '2013-12-27', '2013-12-28',
               '2013-12-30', '2013-12-31'],
              dtype='datetime64[ns]', name='order_date', length=991, freq=None)

In [49]:
#Now let's resample the data into means of monthly sales of furniture and save this into a new variable called Order date
y_furniture = furniture['sales'].resample('MS').mean()
y_furniture

order_date
2011-01-01    1800.615385
2011-02-01    2173.043478
2011-03-01    2239.541667
2011-04-01    1702.346154
2011-05-01    1397.925926
2011-06-01    2612.555556
2011-07-01    1772.269231
2011-08-01    2855.392857
2011-09-01    2853.000000
2011-10-01    2760.800000
2011-11-01    2277.571429
2011-12-01    3907.633333
2012-01-01    1854.750000
2012-02-01    2102.392857
2012-03-01    2398.275862
2012-04-01    2343.148148
2012-05-01    2601.185185
2012-06-01    2775.214286
2012-07-01    1781.370370
2012-08-01    2935.285714
2012-09-01    2857.857143
2012-10-01    2682.066667
2012-11-01    3254.241379
2012-12-01    2936.428571
2013-01-01    2584.033333
2013-02-01    2097.360000
2013-03-01    2532.481481
2013-04-01    2826.222222
2013-05-01    3431.827586
2013-06-01    3698.888889
2013-07-01    2374.655172
2013-08-01    4002.241379
2013-09-01    4032.482759
2013-10-01    3211.535714
2013-11-01    4390.206897
2013-12-01    4449.821429
Freq: MS, Name: sales, dtype: float64

In [50]:
%pip install statsmodels


Note: you may need to restart the kernel to use updated packages.


In [51]:
%pip install --upgrade numpy


  Using cached numpy-1.24.4-cp38-cp38-macosx_10_9_x86_64.whl (19.8 MB)
  Attempting uninstall: numpy
    Found existing installation: numpy 1.22.0
    Uninstalling numpy-1.22.0:
      Successfully uninstalled numpy-1.22.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
scipy 1.6.2 requires numpy<1.23.0,>=1.16.5, but you have numpy 1.24.4 which is incompatible.
Note: you may need to restart the kernel to use updated packages.


In [52]:
%pip install numpy==1.22.0


  Using cached numpy-1.22.0-cp38-cp38-macosx_10_9_x86_64.whl (17.6 MB)
  Attempting uninstall: numpy
    Found existing installation: numpy 1.24.4
    Uninstalling numpy-1.24.4:
      Successfully uninstalled numpy-1.24.4
Note: you may need to restart the kernel to use updated packages.


In [53]:
from statsmodels.tsa.stattools import adfuller

def adf_test(series, title=''):
    """
    Pass in a time series and an optional title, returns an ADF report
    """
    print(f'Augmented Dickey-Fuller Test: {title}')
    result = adfuller(series.dropna(), autolag='AIC')  # .dropna() handles differenced data

    labels = ['ADF test statistic', 'p-value', '# lags used', '# observations']
    out = pd.Series(result[0:4], index=labels)

    for key, val in result[4].items():
        out[f'critical value ({key})'] = val

    print(out.to_string())  # .to_string() removes the line "dtype: float64"

    if result[1] <= 0.05:
        print("Strong evidence against the null hypothesis")
        print("Reject the null hypothesis")
        print("Data has no unit root and is stationary")
    else:
        print("Weak evidence against the null hypothesis")
        print("Fail to reject the null hypothesis")
        print("Data has a unit root and is non-stationary")


adf_test(y_furniture)

Augmented Dickey-Fuller Test: 
ADF test statistic      -0.340857
p-value                  0.919515
# lags used              2.000000
# observations          33.000000
critical value (1%)     -3.646135
critical value (5%)     -2.954127
critical value (10%)    -2.615968
Weak evidence against the null hypothesis
Fail to reject the null hypothesis
Data has a unit root and is non-stationary


In [54]:
office = df.loc[df['category'] == 'Office Supplies']
office['order_date'] = pd.to_datetime(office['order_date'])
office = office.sort_values('order_date')
office = office.groupby('order_date')['sales'].sum().reset_index()
office = office.set_index('order_date')
y_office = office['sales'].resample('MS').mean()
adf_test(y_office)

Augmented Dickey-Fuller Test: 
ADF test statistic      -0.535314
p-value                  0.884939
# lags used              2.000000
# observations          33.000000
critical value (1%)     -3.646135
critical value (5%)     -2.954127
critical value (10%)    -2.615968
Weak evidence against the null hypothesis
Fail to reject the null hypothesis
Data has a unit root and is non-stationary


<ipython-input-54-e3231dd6cb6f>:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [55]:
technology = df.loc[df['category'] == 'Technology']
technology['order_date'] = pd.to_datetime(technology['order_date'])
technology = technology.sort_values('order_date')
technology = technology.groupby('order_date')['sales'].sum().reset_index()
technology = technology.set_index('order_date')
y_technology = technology['sales'].resample('MS').mean()
adf_test(y_technology)

Augmented Dickey-Fuller Test: 
ADF test statistic      -0.146793
p-value                  0.944574
# lags used              8.000000
# observations          27.000000
critical value (1%)     -3.699608
critical value (5%)     -2.976430
critical value (10%)    -2.627601
Weak evidence against the null hypothesis
Fail to reject the null hypothesis
Data has a unit root and is non-stationary


<ipython-input-55-bef5a76a52c1>:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



In [56]:
import plotly.graph_objects as go
fig = go.Figure(data=go.Scatter(x=df.index, y=df['sales'], mode='lines'))
fig.update_layout(
    title="Mức sales theo ngày",
    xaxis_title="Ngày",
    yaxis_title="Sales"
)
fig.show()


In [60]:
df.describe()

sales      quantity      discount        profit  shipping_cost
count  33759.000000  33759.000000  33759.000000  33759.000000   33759.000000
mean     247.130069      3.486181      0.142708     28.581085      26.431853
std      492.653965      2.272437      0.212516    174.499684      57.683970
min        1.000000      1.000000      0.000000  -6599.978000       0.000000
25%       31.000000      2.000000      0.000000      0.000000       2.630000
50%       85.000000      3.000000      0.000000      9.264000       7.820000
75%      251.000000      5.000000      0.200000     36.810000      24.445000
max    22638.000000     14.000000      0.850000   8399.976000     933.570000

**Product Analysis**

In [61]:
profitDF = df['profit'] > 0
profitDF = df[profitDF]
profitPercentage = profitDF.shape[0]/df.shape[0]*100
print('Profit Transactions (profit > 0) :', profitDF.shape[0], 'Transactions Or', "{:.2f}".format(profitPercentage), '%')

lossesDF = df['profit'] < 0
lossesDF = df[lossesDF]
lossesPercentage = lossesDF.shape[0]/df.shape[0]*100
print('Losses Transactions (profit < 0) :', lossesDF.shape[0], 'Transactions Or', "{:.2f}".format(lossesPercentage), '%')

zeroDF = df['profit'] == 0
zeroDF = df[zeroDF]
zeroPercentage = zeroDF.shape[0]/df.shape[0]*100
print('Zero Transactions (profit = 0) :', ' ',zeroDF.shape[0], 'Transactions Or ', "{:.2f}".format(zeroPercentage), '%')

Profit Transactions (profit > 0) : 25083 Transactions Or 74.30 %
Losses Transactions (profit < 0) : 8232 Transactions Or 24.38 %
Zero Transactions (profit = 0) :   444 Transactions Or  1.32 %


In [62]:
product = df.groupby(['product_name', 'category', 'sub_category']).agg({'sales' : 'mean', 'quantity' : 'sum', 'profit' : 'sum'})

In [ ]:
profitProduct = product['profit'] > 0
profitProduct = product[profitProduct]
profitProduct = profitProduct.sort_values('profit', ascending = False)
profitProduct

sales  \
product_name                                       category        sub_category                 
Motorola Smart Phone, Full Size                    Technology      Phones         2280.050000   
Cisco Smart Phone, Full Size                       Technology      Phones         2054.227273   
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       12950.000000   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        2868.857143   
Nokia Smart Phone, Full Size                       Technology      Phones         1681.645161   
...                                                                                       ...   
Tensor "Hersey Kiss" Styled Floor Lamp             Furniture       Furnishings      40.500000   
Kraft Clasp Envelope, Security-Tint                Office Supplies Envelopes        20.875000   
Newell 339                                         Office Supplies Art               4.000000   
Letter Slitter                                     Office Supplies Supplies          5.000000   
Alliance Rubber Bands                              Office Supplies Fasteners         5.000000   

                                                                                 quantity  \
product_name                                       category        sub_category             
Motorola Smart Phone, Full Size                    Technology      Phones              81   
Cisco Smart Phone, Full Size                       Technology      Phones              83   
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers              9   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders             17   
Nokia Smart Phone, Full Size                       Technology      Phones             104   
...                                                                                   ...   
Tensor "Hersey Kiss" Styled Floor Lamp             Furniture       Furnishings          7   
Kraft Clasp Envelope, Security-Tint                Office Supplies Envelopes           18   
Newell 339                                         Office Supplies Art                  2   
Letter Slitter                                     Office Supplies Supplies             4   
Alliance Rubber Bands                              Office Supplies Fasteners            3   

                                                                                     profit  
product_name                                       category        sub_category              
Motorola Smart Phone, Full Size                    Technology      Phones        10719.5560  
Cisco Smart Phone, Full Size                       Technology      Phones         9976.1726  
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers        9519.9728  
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        9278.2270  
Nokia Smart Phone, Full Size                       Technology      Phones         8912.8695  
...                                                                                     ...  
Tensor "Hersey Kiss" Styled Floor Lamp             Furniture       Furnishings       0.5196  
Kraft Clasp Envelope, Security-Tint                Office Supplies Envelopes         0.4000  
Newell 339                                         Office Supplies Art               0.3336  
Letter Slitter                                     Office Supplies Supplies          0.3024  
Alliance Rubber Bands                              Office Supplies Fasteners         0.2016  

[3031 rows x 3 columns]

In [63]:
bestProduct = profitProduct.head(10)
bestProduct

sales  \
product_name                                       category        sub_category                 
Motorola Smart Phone, Full Size                    Technology      Phones         2280.050000   
Cisco Smart Phone, Full Size                       Technology      Phones         2054.227273   
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers       12950.000000   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        2868.857143   
Nokia Smart Phone, Full Size                       Technology      Phones         1681.645161   
Harbour Creations Executive Leather Armchair, A... Furniture       Chairs         1316.035714   
Belkin Router, USB                                 Technology      Accessories     822.000000   
Nokia Smart Phone, with Caller ID                  Technology      Phones         1936.050000   
Hoover Stove, Red                                  Office Supplies Appliances     1954.000000   
Novimex Executive Leather Armchair, Adjustable     Furniture       Chairs         1096.464286   

                                                                                 quantity  \
product_name                                       category        sub_category             
Motorola Smart Phone, Full Size                    Technology      Phones              81   
Cisco Smart Phone, Full Size                       Technology      Phones              83   
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers              9   
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders             17   
Nokia Smart Phone, Full Size                       Technology      Phones             104   
Harbour Creations Executive Leather Armchair, A... Furniture       Chairs             103   
Belkin Router, USB                                 Technology      Accessories         90   
Nokia Smart Phone, with Caller ID                  Technology      Phones              75   
Hoover Stove, Red                                  Office Supplies Appliances          31   
Novimex Executive Leather Armchair, Adjustable     Furniture       Chairs              90   

                                                                                     profit  
product_name                                       category        sub_category              
Motorola Smart Phone, Full Size                    Technology      Phones        10719.5560  
Cisco Smart Phone, Full Size                       Technology      Phones         9976.1726  
Canon imageCLASS 2200 Advanced Copier              Technology      Copiers        9519.9728  
Fellowes PB500 Electric Punch Plastic Comb Bind... Office Supplies Binders        9278.2270  
Nokia Smart Phone, Full Size                       Technology      Phones         8912.8695  
Harbour Creations Executive Leather Armchair, A... Furniture       Chairs         8879.6036  
Belkin Router, USB                                 Technology      Accessories    7048.3200  
Nokia Smart Phone, with Caller ID                  Technology      Phones         6760.8957  
Hoover Stove, Red                                  Office Supplies Appliances     6684.7350  
Novimex Executive Leather Armchair, Adjustable     Furniture       Chairs         6411.5476

*Những sản phẩm đem lại lợi nhuận cao nhất ( đề xuất đẩy mạnh doanh số của những mặt hàng này): Phones, Copiers, Binders, Chairs, Accessories, Appliances*

In [64]:
lossProduct = product['profit'] < 0
lossProduct = product[lossProduct]
lossProduct = lossProduct.sort_values('profit', ascending = True)
lossProduct

sales  \
product_name                                       category        sub_category                
Cubify CubeX 3D Printer Double Head Print          Technology      Machines      3150.000000   
Hoover Stove, White                                Office Supplies Appliances    1747.000000   
Motorola Smart Phone, Cordless                     Technology      Phones        1847.882353   
GBC DocuBind P400 Electric Binding System          Office Supplies Binders       2722.000000   
Bevis Conference Table, Fully Assembled            Furniture       Tables        1994.285714   
...                                                                                      ...   
Pressboard Covers with Storage Hooks, 9 1/2" x ... Office Supplies Binders          5.875000   
Safco Floating Shelf Set, Metal                    Furniture       Bookcases      507.000000   
Cush Cases Heavy Duty Rugged Cover Case for Sam... Technology      Phones          13.333333   
ACCOHIDE Binder by Acco                            Office Supplies Binders         12.000000   
GBC Instant Report Kit                             Office Supplies Binders          9.857143   

                                                                                 quantity  \
product_name                                       category        sub_category             
Cubify CubeX 3D Printer Double Head Print          Technology      Machines             7   
Hoover Stove, White                                Office Supplies Appliances          60   
Motorola Smart Phone, Cordless                     Technology      Phones              60   
GBC DocuBind P400 Electric Binding System          Office Supplies Binders             19   
Bevis Conference Table, Fully Assembled            Furniture       Tables              25   
...                                                                                   ...   
Pressboard Covers with Storage Hooks, 9 1/2" x ... Office Supplies Binders             18   
Safco Floating Shelf Set, Metal                    Furniture       Bookcases           46   
Cush Cases Heavy Duty Rugged Cover Case for Sam... Technology      Phones              11   
ACCOHIDE Binder by Acco                            Office Supplies Binders             11   
GBC Instant Report Kit                             Office Supplies Binders             21   

                                                                                       profit  
product_name                                       category        sub_category                
Cubify CubeX 3D Printer Double Head Print          Technology      Machines     -9.239969e+03  
Hoover Stove, White                                Office Supplies Appliances   -4.195561e+03  
Motorola Smart Phone, Cordless                     Technology      Phones       -3.501924e+03  
GBC DocuBind P400 Electric Binding System          Office Supplies Binders      -3.075837e+03  
Bevis Conference Table, Fully Assembled            Furniture       Tables       -2.748399e+03  
...                                                                                       ...  
Pressboard Covers with Storage Hooks, 9 1/2" x ... Office Supplies Binders      -1.964000e-01  
Safco Floating Shelf Set, Metal                    Furniture       Bookcases    -1.600000e-01  
Cush Cases Heavy Duty Rugged Cover Case for Sam... Technology      Phones       -1.485000e-01  
ACCOHIDE Binder by Acco                            Office Supplies Binders      -1.239000e-01  
GBC Instant Report Kit                             Office Supplies Binders      -8.881784e-16  

[705 rows x 3 columns]

*Có 705 đơn bị profit âm( Cần chú trọng xử lí phần profit bị mất ở đây)*

In [65]:
lossProduct.head(10)

sales  \
product_name                              category        sub_category                
Cubify CubeX 3D Printer Double Head Print Technology      Machines      3150.000000   
Hoover Stove, White                       Office Supplies Appliances    1747.000000   
Motorola Smart Phone, Cordless            Technology      Phones        1847.882353   
GBC DocuBind P400 Electric Binding System Office Supplies Binders       2722.000000   
Bevis Conference Table, Fully Assembled   Furniture       Tables        1994.285714   
Bush Library with Doors, Pine             Furniture       Bookcases      686.750000   
Bevis Round Table, Adjustable Height      Furniture       Tables        1006.000000   
Safco Library with Doors, Traditional     Furniture       Bookcases      577.666667   
Samsung Audio Dock, with Caller ID        Technology      Phones         403.312500   
Office Star Swivel Stool, Adjustable      Furniture       Chairs         475.086957   

                                                                        quantity  \
product_name                              category        sub_category             
Cubify CubeX 3D Printer Double Head Print Technology      Machines             7   
Hoover Stove, White                       Office Supplies Appliances          60   
Motorola Smart Phone, Cordless            Technology      Phones              60   
GBC DocuBind P400 Electric Binding System Office Supplies Binders             19   
Bevis Conference Table, Fully Assembled   Furniture       Tables              25   
Bush Library with Doors, Pine             Furniture       Bookcases           29   
Bevis Round Table, Adjustable Height      Furniture       Tables              11   
Safco Library with Doors, Traditional     Furniture       Bookcases           33   
Samsung Audio Dock, with Caller ID        Technology      Phones              59   
Office Star Swivel Stool, Adjustable      Furniture       Chairs              98   

                                                                           profit  
product_name                              category        sub_category             
Cubify CubeX 3D Printer Double Head Print Technology      Machines     -9239.9692  
Hoover Stove, White                       Office Supplies Appliances   -4195.5612  
Motorola Smart Phone, Cordless            Technology      Phones       -3501.9240  
GBC DocuBind P400 Electric Binding System Office Supplies Binders      -3075.8374  
Bevis Conference Table, Fully Assembled   Furniture       Tables       -2748.3993  
Bush Library with Doors, Pine             Furniture       Bookcases    -2249.2977  
Bevis Round Table, Adjustable Height      Furniture       Tables       -2112.9060  
Safco Library with Doors, Traditional     Furniture       Bookcases    -2105.1738  
Samsung Audio Dock, with Caller ID        Technology      Phones       -2051.9417  
Office Star Swivel Stool, Adjustable      Furniture       Chairs       -1999.4620

*Một số sản phẩm bị profit( cần đề ra hướng giải quyết đoạn này)*

In [66]:
zeroProduct = product['profit'] == 0
zeroProduct = product[zeroProduct]
zeroProduct.head(10)

sales  \
product_name                             category        sub_category               
Alliance Big Bands Rubber Bands, 12/Pack Office Supplies Fasteners       7.500000   
Belkin OmniView SE Rackmount Kit         Office Supplies Storage       213.000000   
Lesro Round Table, Fully Assembled       Furniture       Tables        906.666667   

                                                                       quantity  \
product_name                             category        sub_category             
Alliance Big Bands Rubber Bands, 12/Pack Office Supplies Fasteners           15   
Belkin OmniView SE Rackmount Kit         Office Supplies Storage              6   
Lesro Round Table, Fully Assembled       Furniture       Tables              12   

                                                                       profit  
product_name                             category        sub_category          
Alliance Big Bands Rubber Bands, 12/Pack Office Supplies Fasteners        0.0  
Belkin OmniView SE Rackmount Kit         Office Supplies Storage          0.0  
Lesro Round Table, Fully Assembled       Furniture       Tables           0.0

**Category Analysis**

In [ ]:
category = df.groupby(['category', 'sub_category']).agg({'sales' : 'mean', 'quantity' : 'sum', 'profit' : 'sum','discount':'mean','shipping_cost':'mean'})

In [ ]:
profitCategory = category
profitCategory = profitCategory.sort_values('profit', ascending = False)
profitCategory

sales  quantity        profit  discount  \
category        sub_category                                                 
Technology      Copiers       665.549618      4812  154519.01222  0.118161   
                Phones        516.716331      7934  146059.36450  0.147535   
Furniture       Chairs        442.237719      8302  100725.52590  0.163447   
                Bookcases     606.858689      5400   98705.36900  0.153374   
Office Supplies Appliances    559.047210      3976   88640.08890  0.143030   
Technology      Accessories   242.796278      7226   88032.37760  0.122106   
Office Supplies Storage       224.541365     11188   69444.53770  0.139870   
                Binders        75.058750     14019   50093.35560  0.177662   
Technology      Machines      535.934917      3172   40240.12770  0.174019   
Office Supplies Paper          69.691071      8715   38231.85210  0.111261   
                Art            76.638358     10667   37870.15120  0.114181   
Furniture       Furnishings   122.187856      7530   31576.77630  0.152993   
Office Supplies Envelopes      71.759214      5650   21241.02500  0.128434   
                Supplies       99.550476      5505   15217.85410  0.125879   
                Labels         27.922444      6027    9955.79660  0.120435   
                Fasteners      34.376493      5511    7853.11670  0.135167   
Furniture       Tables        901.007018      2056  -33537.48030  0.290491   

                              shipping_cost  
category        sub_category                 
Technology      Copiers           71.407252  
                Phones            55.537405  
Furniture       Chairs            48.729886  
                Bookcases         62.883679  
Office Supplies Appliances        59.871854  
Technology      Accessories       27.390436  
Office Supplies Storage           23.982171  
                Binders            7.621775  
Technology      Machines          55.755248  
Office Supplies Paper              7.625061  
                Art                8.555202  
Furniture       Furnishings       12.746395  
Office Supplies Envelopes          7.815104  
                Supplies          10.088076  
                Labels             2.958508  
                Fasteners          3.757687  
Furniture       Tables            94.532018

In [ ]:
import plotly.graph_objects as go
fig = go.Figure(data=go.Pie(labels=profitCategory.index, values=df['profit'], textinfo='label+percent', 
                           hoverinfo='value', hole=0.3, sort=False))
fig.show()

**SALES**


* Segment 

In [ ]:
import plotly.express as px
segment_sales = df.groupby('segment')['sales'].sum().reset_index()
fig = px.pie(segment_sales, values='sales', names='segment')
fig.show()

* Market

In [ ]:
market_sales = df.groupby('market')['sales'].sum().reset_index()
fig = px.pie(market_sales, values='sales', names='market')
fig.show()

* Sub-Category

In [ ]:
sub_category_sales = df.groupby('sub_category')['sales'].sum().reset_index()
fig = px.pie(sub_category_sales, values='sales', names='sub_category')
fig.show()

* Quantity theo từng Market

In [ ]:
fig = px.scatter(df, x='sales', y='quantity', color='market')
fig.show()